sdk/trace/batch_span_processor.go

```diff
@@ -120,6 +139,55 @@ func NewBatchSpanProcessor(exporter SpanExporter, options ...BatchSpanProcessorO
 	return bsp
 }
 
+var processorIDCounter atomic.Int64
+
+// nextprocessorID returns an identifier for this batch span processor,
+// starting with 0 and incrementing by 1 each time it is called.
+func nextprocessorID() int64 {
+	return processorIDCounter.Add(1) - 1
+}
```

Comment:
casing (https://github.com/open-telemetry/opentelemetry-go/pull/6393#discussion_r2243096002)

```suggestion
// nextProcessorID returns an identifier for this batch span processor,
// starting with 0 and incrementing by 1 each time it is called.
func nextProcessorID() int64 {
	return processorIDCounter.Add(1) - 1
}
```

Reply:
done